packages/wallet/src/services/util/connectionStatusTracker.ts

```diff
@@ -0,0 +1,40 @@
+import { BehaviorSubject, Observable, distinctUntilChanged, fromEvent, map, merge, shareReplay, startWith } from 'rxjs';
+
+export enum ConnectionStatus {
+  down = 0,
+  up
+}
+
+export type ConnectionStatusTracker = Observable<ConnectionStatus>;
+
+export interface ConnectionStatusTrackerInternals {
+  isNodeEnv?: () => boolean;
```

Comment:
I think these could simply be properties. I'll try to refactor...
I'm changing this PR to draft, but please take a look 😄

Reply:
I have simplified the internals here: [6607665](https://github.com/input-output-hk/cardano-js-sdk/pull/320/commits/6607665dbd8d0208919a3c998227ab9f5f067951)